tools/myst/enc/enc.c

```diff
@@ -139,6 +139,15 @@ static uint64_t _forward_exception_as_signal_to_kernel(
         _mcontext_to_oe_context(&_mcontext, oe_context);
         return OE_EXCEPTION_CONTINUE_EXECUTION;
     }
+    if (oe_exception_code == OE_EXCEPTION_DIVIDE_BY_ZERO)
+    {
+        // ATTN: Are other si_code's possible?
+        siginfo.si_code = FPE_INTDIV;
```

Comment:
The exit info doesn't indicate whether this is a integer divide-by-0 or floating point divide-by-0. I think `FPE_INTDIV` is good enough here. Please update the comment.

Reply:
Thanks for clarifying. What about other SIGFPE cases like overflow, inexact result, out of range etc., will they generate a different `oe_exception_code`?

Reply:
OE code maps x86 #DE (Divide Error Exception) to `OE_EXCEPTION_DIVIDE_BY_ZERO`. #DE is integer divide-by-zero exception. Floating point divided-by-zero exception is part of x86 #MF (`OE_EXCEPTION_X87_FLOAT_POINT`) or #XM (`OE_EXCEPTION_SIMD_FLOAT_POINT`). For accurate [si_code](https://www.man7.org/linux/man-pages/man0/signal.h.0p.html) for SIGFPE, the implementation likely needs to parse the FPU status register or MXCSR register (for SIMD). Please take this opportunity to add "ATTN" comments to the code. 

ATTN: Consider implementing accurate si-code for OE_EXCEPTION_X87_FLOAT_POINT

ATTN: Consider forwarding OE_EXCEPTION_BOUND_OUT_OF_RANGE, OE_EXCEPTION_ACCESS_VIOLATION, OE_EXCEPTION_MISALIGNMENT, OE_EXCEPTION_SIMD_FLOAT_POINT as signal.

Reply:
> ATTN: Consider implementing accurate si-code for OE_EXCEPTION_X87_FLOAT_POINT

Can we replace the si_code to FPE_FLTDIV?

Reply:
> Please take this opportunity to add "ATTN" comments to the code.

Added.

Reply:
@bodzhang Yes we should fill in the accurate `si_code` as long as OE provides the details.

Reply:
@vtikoo 
`FPE_FLTINV`: Invalid floating-point operation
may be a more general option to put for now.
As @bodzhang mentions, we need some work to determine the accurate code, which is captured by the ATTN comment.

Reply:
I think for OE_EXCEPTION_DIVIDE_BY_ZERO, FPE_INTDIV is the right si_code.

Reply:
Yes, I was referring to the OE_EXCEPTION_X87_FLOAT_POINT case